src/Functions/UserDefined/UDFHelper.cpp

```diff
@@ -0,0 +1,285 @@
+#include <Parsers/ASTIdentifier.h>
+#include <Common/filesystemHelpers.h>
+
+#include <Poco/JSON/Parser.h>
+#include <Poco/Util/AbstractConfiguration.h>
+
+#include <boost/algorithm/string/split.hpp>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int BAD_REQUEST_PARAMETER;
+extern const int UDF_INVALID_NAME;
+extern const int FUNCTION_ALREADY_EXISTS;
+extern const int CANNOT_DROP_FUNCTION;
+extern const int UNKNOWN_FUNCTION;
+extern const int UNSUPPORTED_METHOD;
+extern const int UNKNOWN_TYPE;
+}
+
+namespace Streaming
+{
+/// Valid UDF function name is [a-zA-Z0-9_] and has to start with _ or [a-zA-Z]
+void validateUDFName(const String & func_name)
+{
+    if (func_name.empty())
+        throw Exception(ErrorCodes::UDF_INVALID_NAME, "UDF name must not be empy");
+
+    /// At least one alphabetic
+    bool has_alphabetic = false;
+    for (auto ch : func_name)
+    {
+        if (!std::isalnum(ch) && ch != '_')
+            throw Exception(ErrorCodes::UDF_INVALID_NAME, "UDF name can contain only alphabetic, numbers and underscores");
+
+        has_alphabetic = std::isalpha(ch);
+    }
+
+    if (!has_alphabetic)
+        throw Exception(ErrorCodes::UDF_INVALID_NAME, "UDF name shall contain at lest one alphabetic");
+
+    if (!std::isalpha(func_name[0]) && func_name[0] != '_')
+        throw Exception(ErrorCodes::UDF_INVALID_NAME, "UDF name's first char shall be an alphabetic or underscore");
+}
+
+namespace
+{
+
+UserDefinedFunctionConfiguration::FuncType getFuncType(String type_name)
+{
+    UserDefinedFunctionConfiguration::FuncType func_type;
+
+    if (type_name == "executable")
+        func_type = UserDefinedFunctionConfiguration::FuncType::EXECUTABLE;
+    else if (type_name == "remote")
+        func_type = UserDefinedFunctionConfiguration::FuncType::REMOTE;
+    else if (type_name == "javascript")
+        func_type = UserDefinedFunctionConfiguration::FuncType::JAVASCRIPT;
+    else
+        throw Exception(
+            ErrorCodes::BAD_ARGUMENTS, "Wrong user defined function type expected 'executable' or 'remote' actual {}", type_name);
+
+    return func_type;
+}
+}
+
+UserDefinedExecutableFunctionPtr
+createUserDefinedExecutableFunction(ContextPtr context, const std::string & name, const Poco::Util::AbstractConfiguration & config)
+{
+    const String key_in_config = "function";
+    auto get_or_throw = [&](String key) -> String {
+        String value = config.getString("function." + key, "");
+        if (value.empty())
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Missing '{}' for user defined function ", key);
+        return value;
+    };
+
+    String type = get_or_throw("type");
+    UserDefinedFunctionConfiguration::FuncType func_type = getFuncType(type);
+
+    ExternalLoadableLifetime lifetime;
+    if (config.has(key_in_config + ".lifetime"))
+        lifetime = ExternalLoadableLifetime(config, key_in_config + ".lifetime");
+
+    auto init_config = [&](UserDefinedFunctionConfigurationPtr cfg) {
```

Comment:
Can we move these three lambda expressions out as free functions. It will be more readable.